app/assets/javascripts/modules/logger.js

```diff
@@ -0,0 +1,30 @@
+(function () {
+  "use strict";
+  window.GOVUK = window.GOVUK || {};
+
+  function Logger() {
+    if (isDevelopmentEnvironment()) {
+      return console;
+    } else {
+      return silentConsole();
+    }
+  }
+
+  function isDevelopmentEnvironment() {
+    return /\.dev\.gov\.uk$/.test(window.location.hostname);
+  }
+
+  function silentConsole() {
+    var fakeConsole = {};
+
+    $.each(console, function(key) {
+      if (typeof console[key] === 'function') {
+        fakeConsole[key] = function() {};
+      }
+    });
+
+    return fakeConsole;
+  }
+
+  GOVUK.Logger = Logger;
+})();
```

Comment:
It could probably hang about, and if people think there's a real use case for it, then I can remove the deprecation warning altogether. The reason for the deprecation warning was because the code is now promoting explicitly passing custom dimensions to tracking code so you know exactly what you're getting; page-level things should ideally be set in `meta` tags.